src/app/content/components/TableOfContents/index.spec.tsx

```diff
@@ -71,13 +75,45 @@ describe('TableOfContents', () => {
     renderer.act(() => {
       store.dispatch(actions.closeToc());
     });
+
     expect(component.root.findByType(TableOfContents).props.isOpen).toBe(false);
+
     renderer.act(() => {
       store.dispatch(actions.openToc());
     });
     expect(component.root.findByType(TableOfContents).props.isOpen).toBe(true);
   });
 
+  it('focuses when opening', () => {
+    jest.spyOn(reactUtils, 'useMatchMobileMediumQuery')
+      .mockReturnValue(true);
+
+    const {root} = renderToDom(<TestContainer store={store}>
+      <ConnectedTableOfContents />
+    </TestContainer>);
+    const sb = root.querySelector('[data-testid="toc"]')!;
+    const firstTocItem = sb.querySelector('ol > li a, old > li summary') as HTMLElement;
+    const focusSpy = jest.spyOn(firstTocItem as any, 'focus');
+
+    reactDomAct(() => {
+      store.dispatch(actions.closeToc());
+    });
+    reactDomAct(() => {
+      sb?.dispatchEvent(new Event('transitionend'));
+    });
+
+    expect(focusSpy).not.toHaveBeenCalled();
+
+    reactDomAct(() => {
+      store.dispatch(actions.openToc());
+    });
+    reactDomAct(() => {
+      sb?.dispatchEvent(new Event('transitionend'));
+    });
```

Comment:
the trick was that the event listener isn't added until the thing re-renders with the open prop, so you need to have the event dispatch in a separate act call after you update the open state